sphinx_gallery/tests/test_gen_rst.py

```diff
@@ -88,3 +92,36 @@ def test_extract_intro():
             result,
             'This is the description of the example which goes on and on')
         assert_false('second paragraph' in result)
+
+
+def test_pattern_matching():
+    """Test if only examples matching pattern are executed"""
+    examples_dir = tempfile.mkdtemp()
+    gallery_dir = tempfile.mkdtemp()
+
+    gallery_conf = {
+        'fname_pattern': '%s/plot_0*' % examples_dir,
+        'examples_dirs': examples_dir,
```

Comment:
I have not given a lot of thought yet but I would rather test directly to examples in the gallery